src/dali/plugin/input/rest/kraken.py

```diff
@@ -108,8 +108,13 @@ def __init__(
         self.__logger: logging.Logger = create_logger(f"{self.__EXCHANGE_NAME}")
         self._initialize_client()
         self._client.load_markets()
-        self._client.markets_by_id.update({'BSVUSD': {'id': 'BSVUSD', _BASE_ID: 'BSV', _BASE: 'BSV', _QUOTE: 'USD'}})
-        self.base_id_to_base: Dict[str, str] = {value[_BASE_ID]: value[_BASE] for key, value in self._client.markets_by_id.items()}
+        self._client.markets_by_id.update({'BSVUSD': {_ID: 'BSVUSD', _BASE_ID: 'BSV', _BASE: 'BSV', _QUOTE: 'USD'}})
+        self.base_id_to_base: Dict[str, str] = {}
+        for dummy_key, value in self._client.markets_by_id.items():
+            # BUGFIX: Because value comes as a list when in a pytest environment.
+            # Its not obvious why this happens as it is supposed to be a dict.
+            value = value if isinstance(value, dict) else value[0]  # type: ignore
```

Comment:
agreed! :( Not sure why?????

Reply:
What does the `list` look like when you run pytest?
```
LOGGER.debug("markets_by_id: %s", exchange.markets_by_id)
```

Reply:
Its basically the dict inside the list. For example, when I run it in a real world setting, I get: 
`self._client.markets_by_id = {'XBTZUSD': {'baseid': 'XBT' 'base': 'BTC', ...}}`

In pytest, 
`self._client.markets_by_id = {'XBTZUSD': [{'baseid': 'XBT' 'base': 'BTC', ...}]}`

Notice the subtle [] brackets in the pytest case.

Reply:
And the list just has one entry? I'm guessing it is the difference between logging in to the server and not (public API vs. private API). My guess is that the public API gives you multiple aliases since it doesn't know the system you are using.

Reply:
Hmm. Not a bad guess but I thought in general market data was public. The private API is like wallets and ledger.

Reply:
It could be giving the ISO and non-ISO tickers for some reason. It's worth inspecting the whole `list`. It would be good if production code doesn't have work-around like this just for a test. At the very least you could mock the response so that it imitates real-world data. If you need help with that let me know.

Reply:
I guess I could mock, I just didn't want to deal with it. Tbh, the list is a size of 1 last I recall. Also, I'm now on side of thinking it isn't public vs private because in that reverted commit - I had a kraken instance that was without api key (i.e., public only calls able) and it had the same behavior. I had to actually edit the pair_converter tests to deal with the problem (as well as mine). It happening in those tests is what actually helped me narrow it down once I realized it was the same problem. 

My general sentiment is the workaround is bizarre so I'll look into mocking haha (painfully).

Reply:
@macanudo527 @eprbell, For informational purposes here is what the value looks like for key `'XLTCZUSD'` in `markets_by_id` in pytest: 
```
        self._client.load_markets()
        print(self._client.markets_by_id['XLTCZUSD'])
        print(len(self._client.markets_by_id['XLTCZUSD']))

# output
    self.base_id_to_base.update({value[_BASE_ID]: value[_BASE]})
TypeError: list indices must be integers or slices, not str
----------------------------------------------- Captured stdout call -----------------------------------------------
[{'id': 'XLTCZUSD', 'symbol': 'LTC/USD', 'base': 'LTC', 'quote': 'USD', 'settle': None, 'baseId': 'XLTC', 'quoteId': 'ZUSD', 'settleId': None, 'darkpool': False, 'altname': 'LTCUSD', 'type': 'spot', 'spot': True, 'margin': True, 'swap': False, 'future': False, 'option': False, 'active': True, 'contract': False, 'linear': None, 'inverse': None, 'taker': 0.0026, 'maker': 0.0016, 'contractSize': None, 'expiry': None, 'expiryDatetime': None, 'strike': None, 'optionType': None, 'precision': {'amount': 1e-08, 'price': 0.01}, 'limits': {'leverage': {'min': 1.0, 'max': 3.0}, 'amount': {'min': 0.06, 'max': None}, 'price': {'min': 0.01, 'max': None}, 'cost': {'min': None, 'max': None}}, 'info': {'altname': 'LTCUSD', 'wsname': 'LTC/USD', 'aclass_base': 'currency', 'base': 'XLTC', 'aclass_quote': 'currency', 'quote': 'ZUSD', 'lot': 'unit', 'cost_decimals': '5', 'pair_decimals': '2', 'lot_decimals': '8', 'lot_multiplier': '1', 'leverage_buy': ['2', '3'], 'leverage_sell': ['2', '3'], 'fees': [['0', '0.26'], ['50000', '0.24'], ['100000', '0.22'], ['250000', '0.2'], ['500000', '0.18'], ['1000000', '0.16'], ['2500000', '0.14'], ['5000000', '0.12'], ['10000000', '0.1']], 'fees_maker': [['0', '0.16'], ['50000', '0.14'], ['100000', '0.12'], ['250000', '0.1'], ['500000', '0.08'], ['1000000', '0.06'], ['2500000', '0.04'], ['5000000', '0.02'], ['10000000', '0.0']], 'fee_volume_currency': 'ZUSD', 'margin_call': '80', 'margin_stop': '40', 'ordermin': '0.06', 'costmin': '0.5', 'tick_size': '0.01', 'status': 'online', 'long_position_limit': '3000', 'short_position_limit': '3000'}}]
1
```

Reply:
This has been fixed with the last c825b387e00e6436aa079550d545615b5a8e2895